core/src/poh_service.rs

```diff
@@ -121,18 +163,68 @@ impl PohService {
         let mut total_lock_time_ns = 0;
         let mut total_hash_time_ns = 0;
         let mut total_tick_time_ns = 0;
+        let mut try_again_mixin = None;
         loop {
-            num_hashes += hashes_per_batch;
             let should_tick = {
-                let mut lock_time = Measure::start("lock");
-                let mut poh_l = poh.lock().unwrap();
-                lock_time.stop();
-                total_lock_time_ns += lock_time.as_ns();
-                let mut hash_time = Measure::start("hash");
-                let r = poh_l.hash(hashes_per_batch);
-                hash_time.stop();
-                total_hash_time_ns += hash_time.as_ns();
-                r
+                let mixin = if let Some(record) = try_again_mixin {
```

Comment:
The logic inside this got quite complicated. Maybe warrants it's own function to calculate this should_tick result now?